junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -1207,7 +1207,7 @@ public static boolean isMethodPresent(Class<?> clazz, Predicate<Method> predicat
 
 	private static Class<?> loadRequiredParameterType(Class<?> clazz, String methodName, String typeName) {
 		// @formatter:off
-		return tryToLoadClass(typeName)
+		return tryToLoadClass(typeName, clazz.getClassLoader() == null ? ClassLoaderUtils.getDefaultClassLoader() : clazz.getClassLoader())
```

Comment:
I'm not convinced this is actually more correct than before. What if the parameter class is from a different module etc. and should be loaded with a different class loader? If users need to use a special class loader, they could use the findMethod(Class, String, Class...) overload, couldn't they?

Reply:
The methods of a class - including their parameters - form part of the class' definition.
The class is defined and resolved by the classloader that loaded it. The resolution process involves loading all of the classes referenced by this class' definition (including the method parameter types).

If you use another classloader to load the param types, it *may* work (eg, if you use a parent classloader), but it may not. On the other hand, if you use the same classloader that the class itself (and its method signatures) was defined with, it's guaranteed to work.